### PR TITLE
Kanban sample data

### PR DESCRIPTION
**Pull request by pro-odoo**

### CODE REVIEW

File: addons/web/static/src/views/model.js [hunk 1]
```diff
@@ -82,6 +82,7 @@ function getSearchParams(props) {
  * @param {Object} params
  * @param {Object} [options]
  * @param {Function} [options.onUpdate]
+ * @param {boolean} [options.keepFirstGroupLevel=false]
```
Comment (aab-odoo) [inline addons/web/static/src/views/model.js]: fillGroupsWithSampleData
fillEmptyGroupsWithSampleData
enableSampleDataIfAllGroupsEmpty
withSampleDataIfGroupsEmpty

File: addons/web/static/src/views/sample_server.js [hunk 2]
```diff
@@ -624,6 +630,14 @@ export class SampleServer {
...
      */
     _mockWebSearchRead(params) {
+        if (this.existingGroups) {
+            const groups = this.existingGroups;
+            const group = groups[groupNumber++ % groups.length];
+            return {
+                records: group.__data.records,
+                length: group.__data.records.length,
+            };
+        }
```
Comment (aab-odoo) [inline addons/web/static/src/views/sample_server.js]: move to searchReadUnity + check model

File: addons/web/static/src/views/sample_server.js [hunk 3]
```diff
@@ -13,6 +13,8 @@ import { cartesian, sortBy as arraySortBy } from "@web/core/utils/arrays";
 
 class UnimplementedRouteError extends Error {}
 
+let groupNumber = 0;
```
Comment (aab-odoo) [inline addons/web/static/src/views/sample_server.js]: searchReadNumber

File: addons/web/static/src/views/kanban/kanban_controller.js [hunk 4]
```diff
@@ -25,7 +25,44 @@ export class KanbanController extends Component {
     setup() {
         this.actionService = useService("action");
         const { Model, archInfo } = this.props;
-        this.model = useState(useModel(Model, this.modelParams));
```
Comment (ged-odoo) [inline addons/web/static/src/views/kanban/kanban_controller.js:28]:
```js
class KanbanSampleModel extends Model {
}
```

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**